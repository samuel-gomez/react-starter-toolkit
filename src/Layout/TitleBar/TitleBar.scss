@import '@axa-fr/react-toolkit-layout-header/dist/Title/title-bar.scss';

.af-title-bar {
  margin-bottom: 0;
  &--backhome {
    .af-title-bar {
      &__wrapper {
        display: flex;
        justify-content: flex-start;
        align-items: center;
        .af-title-bar__title {
          order: 2;
          margin-left: 1em;
        }
      }
    }
  }
}

@include media-breakpoint-down(sm) {
  .af-title-bar {
    position: sticky;
    width: 100%;
    top: 69px;
    z-index: 8;
    &__title {
      font-size: 1.2rem;
    }
    .af-btn--circle {
      width: 2rem;
      height: 2rem;
    }
  }
}
