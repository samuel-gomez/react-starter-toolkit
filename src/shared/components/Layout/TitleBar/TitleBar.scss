@import '@axa-fr/react-toolkit-layout-header/dist/Title/title-bar.scss';
@import 'shared/scss/variables.scss';

.af-title-bar {
  margin-bottom: 0;
  &--backhome {
    .af-title-bar {
      &__wrapper {
        display: flex;
        justify-content: flex-start;
        align-items: center;
        .af-title-bar__title {
          order: 2;
          margin-left: 1em;
        }
      }
    }
  }
  &--fullscreen {
    .container {
      width: 100%;
    }
  }
}

@include media-breakpoint-down(sm) {
  .af-title-bar {
    position: sticky;
    width: 100%;
    top: calc(var(--fullheaderHeight) - var(--titlebarHeight));
    z-index: 8;
    &__title {
      font-size: 1.2rem;
    }
    .af-btn--circle {
      width: 2rem;
      height: 2rem;
    }
  }
}
