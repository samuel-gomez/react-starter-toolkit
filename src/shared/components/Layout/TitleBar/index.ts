export { default } from './TitleBar.container';
