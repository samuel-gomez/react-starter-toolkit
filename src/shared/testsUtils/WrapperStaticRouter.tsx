/* eslint-disable react/display-name */
import { ReactNode } from 'react';
import { MemoryRouterProvider } from 'next-router-mock/MemoryRouterProvider';

const WrapperStaticRouter =
  (route = '/') =>
  ({ children }: { children: ReactNode }) =>
    <MemoryRouterProvider url={route}>{children}</MemoryRouterProvider>;

export default WrapperStaticRouter;
