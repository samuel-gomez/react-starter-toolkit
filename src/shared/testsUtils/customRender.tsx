/* eslint-disable react/display-name */
import { render, RenderOptions } from '@testing-library/react';
import { MemoryRouterProvider } from 'next-router-mock/MemoryRouterProvider';

import FetchProvider from 'providers/FetchProvider';
import UserProvider from 'providers/UserProvider';
import { ReactElement, ReactNode } from 'react';
import { MOCK_API_URL } from '.';

type TMockProvider = {
  [x: string]: Record<string, unknown | number | string> | string;
};

const MockProviders =
  ({ role = '', ...testMock }: TMockProvider) =>
  ({ children }: { children: ReactNode }) => {
    const useOidcAccessTokenFn = jest.fn().mockReturnValue({ accessToken: 'accessToken' });
    const useOidcUserFn = jest.fn().mockReturnValueOnce({ oidcUser: { member_of: [`CN=${role}`], name: 'Samuel Gomez' } });
    return (
      <UserProvider useOidcUserFn={useOidcUserFn}>
        <FetchProvider
          apiUrl={MOCK_API_URL}
          fetchConfig={{
            headers: { testMock: JSON.stringify(testMock) },
          }}
          useOidcAccessTokenFn={useOidcAccessTokenFn}
        >
          <MemoryRouterProvider url="">{children}</MemoryRouterProvider>
        </FetchProvider>
      </UserProvider>
    );
  };

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>, testMock: TMockProvider = { role: '' }) =>
  render(ui, { wrapper: MockProviders(testMock), ...options });

// re-export everything
export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

// override render method
export { customRender as render };
