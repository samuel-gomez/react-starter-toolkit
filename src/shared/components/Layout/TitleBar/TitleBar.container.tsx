import { ComponentProps } from 'react';
import TitleBar, { DEFAULT_CLASSNAME } from './TitleBar';

type TTitleBarContainer = ComponentProps<typeof TitleBar> & {
  TitleBarCmpt?: typeof TitleBar;
  fullScreen?: boolean;
};

const TitleBarContainer = ({ fullScreen, backHome = false, classModifier = '', className = DEFAULT_CLASSNAME, ...rest }: TTitleBarContainer) => {
  const newClassModifier = [classModifier, backHome ? 'backhome' : '', fullScreen ? 'fullscreen' : ''].join(' ').trim();
  return <TitleBar {...rest} backHome={backHome} classModifier={newClassModifier} className={className} />;
};

export default TitleBarContainer;
