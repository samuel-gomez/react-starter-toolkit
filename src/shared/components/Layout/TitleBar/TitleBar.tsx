import { ReactNode } from 'react';
import Link from 'next/link';
import { Title as HeaderTitle } from '@axa-fr/react-toolkit-layout-header';
import { withClassDefault, withClassModifier, WithClassModifierOptions, compose, identity } from '@axa-fr/react-toolkit-core';
import './TitleBar.scss';

type TTitleBar = WithClassModifierOptions & {
  backHome?: boolean;
  children?: ReactNode;
  className?: string;
  title?: string;
  handleClick?: () => void;
};

export const DEFAULT_CLASSNAME = 'af-title-bar';

const TitleBar = ({ title = 'title', handleClick, children = <></>, backHome = false, className }: TTitleBar) => (
  <HeaderTitle className={className} toggleMenu={handleClick} title={title}>
    {backHome && (
      <Link aria-label="Retour Ã  l'accueil" className="btn af-btn--circle" href="/">
        <i className="glyphicon glyphicon-home" />
      </Link>
    )}
    {children}
  </HeaderTitle>
);

const enhance = compose(identity<TTitleBar>(), withClassDefault(DEFAULT_CLASSNAME), withClassModifier());

export default enhance(TitleBar);
