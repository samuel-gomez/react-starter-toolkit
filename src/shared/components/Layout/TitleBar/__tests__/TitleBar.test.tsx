import { render, screen } from '@testing-library/react';
import { emptyFunction, renderWithWrapperStaticRouter } from 'shared/testsUtils';
import TitleBar from '../TitleBar';

const defaultProps = { title: 'title', handleClick: emptyFunction };

describe('<TitleBar/>', () => {
  it('Render <TitleBar />', () => {
    const { asFragment } = render(<TitleBar {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Render <TitleBar /> with other classname and child', () => {
    const { asFragment, getByText } = render(
      <TitleBar {...defaultProps} className="other">
        <p>child</p>
      </TitleBar>,
    );
    expect(getByText('child')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should contain back home <Link /> when backHome is true', () => {
    const { asFragment } = renderWithWrapperStaticRouter(<TitleBar {...defaultProps} backHome />);
    expect(screen.getByLabelText("Retour Ã  l'accueil")).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });
});
