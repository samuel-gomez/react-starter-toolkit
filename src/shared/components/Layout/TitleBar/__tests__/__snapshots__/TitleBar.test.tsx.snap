// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleBar/> Render <TitleBar /> 1`] = `
<DocumentFragment>
  <div
    class="af-title-bar"
  >
    <div
      class="container af-title-bar__wrapper"
    >
      <div
        class="burger-container"
      >
        <a
          aria-controls="mainmenu"
          aria-haspopup="true"
          class="btn af-title-bar__mobile-menu af-btn--circle"
          href="/#"
          id="togglemenu"
          role="button"
          tabindex="0"
          title="Toggle menu"
        >
          <i
            class="glyphicon glyphicon-menu-hamburger"
          />
        </a>
      </div>
      <h1
        class="af-title-bar__title"
      >
        title
      </h1>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<TitleBar/> Render <TitleBar /> with other classname and child 1`] = `
<DocumentFragment>
  <div
    class="other"
  >
    <div
      class="container af-title-bar__wrapper"
    >
      <div
        class="burger-container"
      >
        <a
          aria-controls="mainmenu"
          aria-haspopup="true"
          class="btn af-title-bar__mobile-menu af-btn--circle"
          href="/#"
          id="togglemenu"
          role="button"
          tabindex="0"
          title="Toggle menu"
        >
          <i
            class="glyphicon glyphicon-menu-hamburger"
          />
        </a>
      </div>
      <h1
        class="af-title-bar__title"
      >
        title
      </h1>
      <p>
        child
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<TitleBar/> Should contain back home <Link /> when backHome is true 1`] = `
<DocumentFragment>
  <div
    class="af-title-bar"
  >
    <div
      class="container af-title-bar__wrapper"
    >
      <div
        class="burger-container"
      >
        <a
          aria-controls="mainmenu"
          aria-haspopup="true"
          class="btn af-title-bar__mobile-menu af-btn--circle"
          href="/#"
          id="togglemenu"
          role="button"
          tabindex="0"
          title="Toggle menu"
        >
          <i
            class="glyphicon glyphicon-menu-hamburger"
          />
        </a>
      </div>
      <h1
        class="af-title-bar__title"
      >
        title
      </h1>
      <a
        aria-label="Retour Ã  l'accueil"
        class="btn af-btn--circle"
        href="/"
      >
        <i
          class="glyphicon glyphicon-home"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;
